### PR TITLE
Allow payment via task

### PR DESCRIPTION
**Pull request by lawyu89**
### Why

_few (or many) words to explain the very purpose of this PR_

### Ticket

https://app.shortcut.com/capable/story/<id>

### Accomplishments

- [x] Did this thing
- [x] Did this other import thing
- [ ] ...

### Environment variables

If you add, change or remove an environment variable, make sure to do the following:

- [ ] Update [Render](https://dashboard.render.com/env-group/evg-c9l8mmp0gd0a70f4b1ng)
- [ ] Update [Notion](https://www.notion.so/capablehealth/Capable-Care-Environment-Variables-INTERNAL-07cbae482a424df8bf3c7bcc2ced74c7)
- [ ] Update 1Password
- [ ] Update .env.sample

### Feature flag

- [ ] If there is a way to toggle the work in this PR via a feature flag, I have added one
- [ ] The flag has been created in [LaunchDarkly](https://app.launchdarkly.com/kariko/dev/features) for all environments

### CODE REVIEW

File: src/features/tasks/pages/Task.tsx [hunk 1]
```diff
@@ -1,23 +1,76 @@
...
+import { HeaderImage, RichText } from "components";
 import { TaskAchievementStatus } from "../models/Task";
 import { useTask } from "../hooks/useTask.hook";
 import { useUpdateTask } from "../hooks/useUpdateTask.hook";
 import ActionButton from "components/ActionButton";
 import useCRMContent from "fetchDataHooks/useCRMContent";
+import { SubscriptionPayment } from "pages/Subscriptions";
+import { useCurrentPatient, useExistingPaymentMethod } from "fetchDataHooks/index";
+import { SubscriptionOption } from "models/subscriptions/Subscription.types";
+import useSubscriptionPlan from "fetchDataHooks/useSubscriptionPlans";
```
Comment (LUMacKenn) [inline src/features/tasks/pages/Task.tsx:15]:
```suggestion
import useSubscriptionPlans from "fetchDataHooks/useSubscriptionPlans";
```

File: src/features/tasks/pages/Task.tsx [hunk 2]
```diff
@@ -1,23 +1,76 @@
...
   const { isLoading: taskWithCRMLoading, isError: taskWithCRMError } = useCRMContent(task);
   const { mutate: updateTask } = useUpdateTask();
-
+  const stripeEnabled =
+    !!process.env.REACT_APP_STRIPE_PUBLIC_KEY && !!process.env.REACT_APP_STRIPE_ACCOUNT_ID;
   const isTaskCompleted = task?.achievement_status === TaskAchievementStatus.COMPLETED;
+  // Payment related
+  // currently there is no way to fetch subscription plans based on product id so well need to subscriptions and filter ourselves
+  const navigate = useNavigate();
+  const { subscriptionPlans, isLoading: plansLoading } = useSubscriptionPlan();
```
Comment (LUMacKenn) [inline src/features/tasks/pages/Task.tsx:29]:
```suggestion
  const { subscriptionPlans, isLoading: plansLoading } = useSubscriptionPlans();
```

File: src/features/tasks/pages/Task.tsx [hunk 3]
```diff
@@ -1,23 +1,76 @@
...
+  const handleSubmit = async () => {
+    if (!task) return;
+
+    await updateTask({
+      taskId: task.id,
+      task: {
+        achievement_status: TaskAchievementStatus.COMPLETED,
+      },
+    });
+    navigate(`/home/${carePlanId}/tasks`);
```
Comment (LUMacKenn) [inline src/features/tasks/pages/Task.tsx:68]: you could put this in an `onSuccess` or `onSettled` callback if you want, and wouldn't need the `await`

**Review by LUMacKenn — APPROVED** [review]
Code looks good, just left a few comments for clarity. But I'm a bit confused, are we using tasks for patients to use subscription plans somehow? How do those overlap?

**Comment by achidu** [comment]
https://app.shortcut.com/capable/story/8513/tasks-support-a-payment

**Comment by lawyu89** [comment]
> 

its weird because currently the only form of payment we can support are recurring subscriptions so were having customer use "one-time" subscriptions to mimic one-time payments

**Pull request merged**